### PR TITLE
3.get/api

### PR DESCRIPTION
**Pull request by rachelkwok00**

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -3,16 +3,26 @@ const app = express();
...
 } = require('./controllers/controller.js');
 
 
 app.get('/api/topics', getTopics);
-app.get('/api', addToApi);
+app.get('/api', getData);
+
+app.all('/*', (req, res, next) => {
+  res.status(404).send({msg: "Invalid api endpoint"});
+});
```
Comment (Augs0) [inline app.js:15]: Nice! However, I can't seem to see a test for this 404. Could you add a test trying to access an endpoint that does not exist and checking that the status code and message match what you have in this line?

File: app.js [hunk 2]
```diff
@@ -3,16 +3,26 @@ const app = express();
...
 
 app.use((err, req, res, next) => {
     console.error(err);
     res.status(500).send({ msg: 'Internal Server Error' });
   });
   
+  app.use((err, req, res, next) => {
+    if (err === "Invalid query") {
```
Comment (Augs0) [inline app.js]: We don't have to worry about queries for this endpoint as none are accepted. The user can only go to /api if they go to a non-existing endpoint, it should hit your 404 error handler. You're very close here in how we will handle errors later though 😄

**Review by Augs0 — CHANGES_REQUESTED** [review]
This is looking good, you're almost there ❇️  I'd like to see a test for the 404 error handler and I'd advise you remove the 400 error handler for now as it's not needed for this endpoint. Please make those changes and resubmit.

**Review by NRMSMITH — APPROVED** [review]
Good work on following August's advice. Free to merge 🚀

**Pull request merged**